RHDEVDOCS-3226 Logging 5.2 docs aggregated branch (#35445)
openshift/openshift-docs: modules/cluster-logging-collector-log-forward-es.adoc

<3> Specify a name for the output.
<4> Specify the `elasticsearch` type.
<5> Specify the URL and port of the external Elasticsearch instance as a valid absolute URL. You can use the `http` (insecure) or `https` (secure HTTP) protocol. If the cluster-wide proxy using the CIDR annotation is enabled, the output must be a server name or FQDN, not an IP Address.
<6> For a secure connection, you can specify an `https` or `http` URL that you authenticate by specifying a `secret`.

> <7> For an `https` prefix, specify the name of the secret required by the endpoint for TLS communication. The secret must exist in the `openshift-logging` project and must have keys of: *tls.crt*, *tls.key*, and *ca-bundle.crt* that point to the respective certificates that they represent. Otherwise, for `http` and `https` prefixes, you can specify a secret that contains a username and password. For more information, see the following "Example: Setting secret that contains a username and password."

Author: @igor-karpukhin WDYT?

Author: @abrennan89, please peer review.

> <7> For an `https` prefix, specify the name of the secret required by the endpoint for TLS communication. The secret must exist in the `openshift-logging` project and must have keys of: *tls.crt*, *tls.key*, and *ca-bundle.crt* that point to the respective certificates that they represent. Otherwise, for `http` and `https` prefixes, you can specify a secret that contains a username and password. For more information, see the following "Example: Setting secret that contains a username and password."

Reviewer: ```suggestion
<7> For an `https` prefix, specify the name of the secret required by the endpoint for TLS communication. The secret must exist in the `openshift-logging` project, and must have keys of: *tls.crt*, *tls.key*, and *ca-bundle.crt* that point to the respective certificates that they represent. Otherwise, for `http` and `https` prefixes, you can specify a secret that contains a username and password. For more information, see the following "Example: Setting secret that contains a username and password."
```

Author: Updated. Thanks.

<8> Optional: Specify a name for the pipeline.

> <9> Specify which log types should be forwarded using that pipeline: `application,` `infrastructure`, or `audit`.

Reviewer: ```suggestion
<9> Specify which log types should be forwarded by using that pipeline: `application,` `infrastructure`, or `audit`.
```

Author: Updated. Thanks.

> <10> Specify the output to use with that pipeline for forwarding the logs.

Reviewer: ```suggestion
<10> Specify the output to use when forwarding logs with this pipeline.
```

Author: Updated. Thanks.

<11> Optional: Specify the `default` output to send the logs to the internal Elasticsearch instance.

> <12> Optional: Forward structured JSON log entries as JSON objects in the `structured` field. The log entry must contain valid structured JSON; otherwise, OpenShift Logging removes the `structured` field and instead sends the log entry to the default index, `app-00000x`.

Reviewer: ```suggestion
<12> Optional: Specify whether to forward structured JSON log entries as JSON objects in the `structured` field. The log entry must contain valid structured JSON; otherwise, OpenShift Logging removes the `structured` field and instead sends the log entry to the default index, `app-00000x`.
```

I'm not sure if you want to maybe keep these consistent, i.e. use specify at the start of all lines?

Author: Updated. Thanks.

<13> Optional: String. One or more labels to add to the logs.
<14> Optional: Configure multiple outputs to forward logs to other external log aggregators of any supported type:

> ** Optional. A name to describe the pipeline.

Reviewer: I'm not sure if Optional needs to be here twice since it's mentioned in the first line? I know it's not part of this PR so feel free to clean up another time, but I'd remove "Optional." here from the first sub bullet

Author: Updated. Thanks.

<3> Specify a name for the output.
<4> Specify the `elasticsearch` type.
<5> Specify the URL and port of the external Elasticsearch instance as a valid absolute URL. You can use the `http` (insecure) or `https` (secure HTTP) protocol. If the cluster-wide proxy using the CIDR annotation is enabled, the output must be a server name or FQDN, not an IP Address.

> <6> For a secure connection, you can specify an `https` or `http` URL that you authenticate by specifying a `secret`.

Author: @anpingli. I updated line 68 and 69 for RHDEVDOCS-2930

Reviewer 2: Looks good

----
$ oc delete pod --selector logging-infra=fluentd
----

> 

Author: @anpingli. I added this section for @anpingli RHDEVDOCS-2930.

Author: @SNiemann15

----
$ oc delete pod --selector logging-infra=fluentd
----

> .Example: Setting a secret that contains a username and password

Author: @igor-karpukhin WDYT?

Reviewer 3: @rolfedh looks good!

Author: @abrennan89, please peer review.

You can use a secret that contains a username and password to authenticate a secure connection to an external Elasticsearch instance.

For example, if you cannot use mutual TLS (mTLS) keys because a third party operates the Elasticsearch instance, you can use HTTP or HTTPS and set a secret that contains the username and password.

. Create a `Secret` YAML file similar to the following example that contains base64 encoded values for `username` and `password`:
+
[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: openshift-test-secret
data:
  username: dGVzdHVzZXJuYW1lCg==
  password: dGVzdHBhc3N3b3JkCg==
----

. Create the secret:
+
[source,terminal]
----
$ oc create secret -n openshift-logging openshift-test-secret.yaml
----

> 

Reviewer: extra blank line, again might be a personal style choice 🙂 up to you

Author: Updated. Thanks.

> . Specify the name of the secret in your `ClusterLogForwarder`:

Reviewer: ```suggestion
. Specify the name of the secret in the `ClusterLogForwarder` resource:
```
General OCP rule is to have a word like "resource" or "object" after

Author: Updated. Thanks.

+
[source,yaml]
----
kind: ClusterLogForwarder
metadata:
  name: instance
  namespace: openshift-logging
spec:
  outputs:
   - name: elasticsearch
     type: "elasticsearch"
     url: https://elasticsearch.secure.com:9200
     secret:
        name: openshift-test-secret
----

[NOTE]
====
With this type of secret, the prefix of the `url` value can be `http` or `https`.

> ====

Reviewer: I'd maybe move this to directly below the YAML for the secret itself?

Author: I think this location is best. This note is intended to help users understand the `url` field in the snippet that precedes the note. I don't want them to infer from this snippet that the value of `url` must only be `https...`  Perhaps there's a better way to convey this. What do you think?

Reviewer: Maybe just a call-out either, but I see what you mean now on second look, so I think it's fine either way 🙂
